dataService.js

```diff
@@ -0,0 +1,70 @@
+let combinedProductMap = null
+
+class DataService {
+  #categoryMap = null;
+  #productMap = null;
+  #allProducts = null;
+  #allCategories = null;
+
+  /*
+   * Map individual product objects to product id attributes
+   */
+  getProducts () {
+    // Since logical terminates on first instance of truthy value, need mot worry
+    // about querying size of null
+    if (!this.#productMap || !this.#productMap.size()) {
```

Comment:
Took advantage of left to right associativity of logical operators, hence size is queried only if variable is not null